File: oblig1/hovedprogram.py
"""
5. Hovedprogram
Lag et Datasenter som leser inn informasjon om regneklyngene
saga og abel fra testfiler som beskrevet ovenfor.

For hver av regneklyngene skal du deretter skrive ut:
navnet på regneklyngen, antall racks, antall prosessorer og
hvor mange noder som har minst 32 GB, 64 GB og 128 GB minne.

Utskriften kan f.eks. se slik ut:
Informasjon om regneklyngen Saga
Antall rack: 56
Antall prosessorer: 682
Noder med minst 32 GB: 666
Noder med minst 64 GB: 666
Noder med minst 128 GB: 16
"""
from datasenter import Datasenter
from regneklynge import Regneklynge
from rack import Rack
from node import Node

abelFilnavn = "abel.txt"
sagaFilnavn = "saga.txt"
abel = abelFilnavn.replace(".txt", "")
saga = sagaFilnavn.replace(".txt", "")

datasenter11 = Datasenter()

datasenter11.lesInnRegneklynge(abelFilnavn, 12)
datasenter11.lesInnRegneklynge(sagaFilnavn, 12)

print("\nInformasjon om regneklyngen: abel")
print("Antall rack:", datasenter11._datasenter[abel].antRacks())
print("Antall prosessorer:", datasenter11._datasenter[abel].antProsessorer())
print("Noder med minst 32 GB: ", datasenter11._datasenter[abel].noderMedNokMinne(32))
print("Noder med minst 64GB: ", datasenter11._datasenter[abel].noderMedNokMinne(64))
print("Noder med minst 128 GB: ", datasenter11._datasenter[abel].noderMedNokMinne(128))

print("\nInformasjon om regneklyngen: saga")
print("Antall rack:", datasenter11._datasenter[saga].antRacks())
print("Antall prosessorer:", datasenter11._datasenter[saga].antProsessorer())
print("Noder med minst 32 GB: ", datasenter11._datasenter[saga].noderMedNokMinne(32))
print("Noder med minst 64GB: ", datasenter11._datasenter[saga].noderMedNokMinne(64))
print("Noder med minst 128 GB: ", datasenter11._datasenter[saga].noderMedNokMinne(128))

File: oblig1/Hovedprogram.java
// IN1010 vår 22 oblig 1: Dataklynge
// Klasse for representasjon av et datasenter

import java.util.Scanner;
import java.io.File;
import java.io.FileNotFoundException;

class Hovedprogram {
    public static void main(String[] args) {
        
        // Oppretter et datasenter, angir dataklynge-filnavn
        Dataklynge saga = new Dataklynge(12);
        String filnavn = "dataklynge2.txt";
        Scanner sc = null;
    
        try {
            sc = new Scanner(new File(filnavn));
        }
        // fanger feilen og terminerer dersom ikke filen eksisterer
        catch (FileNotFoundException e) {
            System.out.println("OBS: finner ikke filen");
        }

        // Leser inn data om en dataklynge fra fil
        // @param filnavn filene der dataene for dataklyngen ligger
        while (sc.hasNextLine()) {
            String[] biter = sc.nextLine().split(" ");

            int antNoder = Integer.parseInt(biter[0]);
            int antProsPrNode = Integer.parseInt(biter[1]);
            int minnePrNode = Integer.parseInt(biter[2]);

            // setter inn Noder etter angitte spesifikasjoner i tekstfilen
            for (int i = 0; i < antNoder; i++){
                saga.settInnNode(new Node(minnePrNode, antProsPrNode)); // type1 node
            }
        }
        // sjekker antall Noder med gitt minste minnestørrelse
        int[] sjekkMinner = {128, 512, 1024};
            for (int i : sjekkMinner) {
                System.out.print("Noder med minst " + i + " GB: ");
                System.out.print(saga.noderMedNokMinne(i) + "\n");
            }
            System.out.println("Antall prosessorer: " + saga.antProsessorer());
            System.out.println("Antall rack i bruk: " + saga.antRacks());

        sc.close();

    }
}
